c++.html.markdown

```diff
@@ -55,24 +55,26 @@ int main(int argc, char** argv)
 
 // However, C++ varies in some of the following ways:
 
-// In C++, character literals are chars
-sizeof('c') == sizeof(char) == 1
+// In C++, character literals are chars, therefore the size is 1
+sizeof('c') == sizeof(char)
 
-// In C, character literals are ints
+// In C, character literals are ints, therefore the size is 4
```

Comment:
Yeah no

Reply:
I mean it is in practice

```c
#include <stdio.h>

int main() {
    printf("Size of character literal 'c': %zu bytes\n", sizeof('c'));
    printf("Size of int: %zu bytes\n", sizeof(int));
    printf("Size of char: %zu bytes\n", sizeof(char));

    return 0;
}
```

```sh
gcc -o test_size test_size.c
./test_size
```

```
Size of character literal 'c': 4 bytes
Size of int: 4 bytes
Size of char: 1 bytes
```

Reply:
In practice for you

Reply:
@vendethiel In C character literals are _always_ 4 bytes in size, where the size of a "char" is 1 byte, on almost every platform. This is mainly because of the constraints of early days when C was developed, and there are no exact reasons for this. C++ and probably any modern languages addresses this. Due to backwards compatibility we still have this.

In good C code, you wouldn't find anyone is using "sizeof('a')", as in C++ this would break.

A better statement there would be: "In C, character literals are treated as ints, therefore each character size is  4  bytes."